File: 151004/Bordilovsky/DC_NoSQL/Publisher/Repositories/IRepository.cs
namespace DC_REST.Repositories
{
	public interface IRepository<TEntity>
	{
		TEntity GetById(int id);
		List <TEntity> GetAll();
		TEntity Add(TEntity entity);
		TEntity Update(int id, TEntity entity);
		bool Delete(int id);
		int GetCurrentId();
	}
}

File: 151002/Samusenko/lab_five/publisher/src/main/java/com/example/publisher/dao/IRepository.java
package com.example.publisher.dao;

import java.util.Optional;

public interface IRepository<T, A> {
    Optional<A> getById(T id);
    Iterable<A> getAll();
    Optional<A> save (A entity);
    Optional<A> update(A entity);
    boolean remove(A entity);
    boolean removeById(T id);
}